File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/DiagnosticReport.ts
import { Attachment } from "./Attachment";
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type DiagnosticReportMedia = BackboneElement & {
    link: Reference;
    comment?: string;
};

export type DiagnosticReport = DomainResource & {
    category?: CodeableConcept[];
    conclusionCode?: CodeableConcept[];
    conclusion?: string;
    encounter?: Reference;
    specimen?: Reference[];
    resultsInterpreter?: Reference[];
    status: string;
    result?: Reference[];
    effective?: string | Period;
    code: CodeableConcept;
    identifier?: Identifier[];
    issued?: string;
    presentedForm?: Attachment[];
    basedOn?: Reference[];
    imagingStudy?: Reference[];
    media?: DiagnosticReportMedia[];
    subject?: Reference;
    performer?: Reference[];
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/DiagnosticReport.py
from typing import Optional, List
from pydantic import *
from ..base import *

class DiagnosticReport_Media(BackboneElement):
    link: Reference
    comment: Optional[str] = None

class DiagnosticReport(DomainResource):
    category: Optional[List[CodeableConcept]] = None
    conclusion_code: Optional[List[CodeableConcept]] = None
    conclusion: Optional[str] = None
    encounter: Optional[Reference] = None
    specimen: Optional[List[Reference]] = None
    effective_date_time: Optional[str] = None
    results_interpreter: Optional[List[Reference]] = None
    status: str
    result: Optional[List[Reference]] = None
    code: CodeableConcept
    identifier: Optional[List[Identifier]] = None
    issued: Optional[str] = None
    presented_form: Optional[List[Attachment]] = None
    based_on: Optional[List[Reference]] = None
    imaging_study: Optional[List[Reference]] = None
    media: Optional[List[DiagnosticReport_Media]] = None
    subject: Optional[Reference] = None
    performer: Optional[List[Reference]] = None
    effective_period: Optional[Period] = None